request/request.go

```diff
@@ -192,32 +192,32 @@ func (r *Request) Size() int {
 }
 
 // SizeAndDo adds an OPT record that the reflects the intent from request.
-// The returned bool indicated if an record was found and normalised.
+// The returned bool indicates if an record was found and normalised.
 func (r *Request) SizeAndDo(m *dns.Msg) bool {
 	o := r.Req.IsEdns0()
 	if o == nil {
 		return false
 	}
 
-	odo := o.Do()
-
 	if mo := m.IsEdns0(); mo != nil {
 		mo.Hdr.Name = "."
 		mo.Hdr.Rrtype = dns.TypeOPT
 		mo.SetVersion(0)
 		mo.SetUDPSize(o.UDPSize())
 		mo.Hdr.Ttl &= 0xff00 // clear flags
 
+		// If the request had options set, but we don't support those, we should not echo those back.
 		if len(o.Option) > 0 {
-			o.Option = supportedOptions(o.Option)
+			mo.Option = supportedOptions(o.Option)
```

Comment:
This code may erase existing options from response OPT record which seems incorrect. I would prefer to not merge this. The fix merged recently has noop code, but it at least doesn't break anything. 

Please consider proposals below
1. Just remove `if len(o.Option) > 0 {` block here. Since response message already has OPT record we can assume that other service (e.g. upstream server or another plugin) already set all options as needed
2. Add options supported by coredns to existing options (perhaps supported by upstream server) in response OPT record, i.e. add code like
```
if len(o.Option) > 0 {
  mo.Option = MERGE(mo.Option, supportedOptions(o.Option)...)
}
```
here `MERGE()` should add supported options but avoid duplicates